controllers/component_build_controller_pac.go

```diff
@@ -148,6 +156,162 @@ func (r *ComponentBuildReconciler) ProvisionPaCForComponent(ctx context.Context,
 	return mrUrl, nil
 }
 
+func getHttpClient() *http.Client {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: gp.IsInsecureSSL()},
+	}
+	client := &http.Client{Transport: tr}
+	return client
+}
+
+// ensureIncomingSecret is ensuring that incoming secret for PaC trigger exists
+// if secret doesn't exists it will create it and also add repository as owner
+// Returns:
+// pointer to secret object
+// string with secret name
+// bool which indicates if reconcile is required (which is required when we just created secret)
```

Comment:
What's the case that just creates the secret but does not trigger a build?

Reply:
either first time when trigger is called, or when secret is removed

we have to re-create secret, BUT we would also have to get the secret (and there is race), so I would have to add sleep there, so Mykola recommended to run reconcile to sort out creation of objects

Reply:
then, how about return an error if the incoming secret cannot be ensured, so that that can be logged or even an event to be emitted?

Reply:
error is returned if there is any issue otherwise error isn't returned